### PR TITLE
1519 staging email refactor

### PR DESCRIPTION
**Pull request by hem-brahmbhatt**
#### Background

- Adds a recipient email address to Staging
- Prefixes the subject of the emails with the environment that the alert was triggered on. This is because all the emails get sent to the same address.
- Refactors config around MECA email alerts

#### Any relevant tickets

Relates to #1519 

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [x] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

### CODE REVIEW

**Review by Cooryd — APPROVED** [review]

File: server/meca/routes.js [hunk 1]
```diff
@@ -38,17 +38,17 @@ module.exports = app => {
     const updateStatus = () => Manuscript.updateStatus(manuscriptId, status)
 
     const sendEmail = () => mailer.send({
-      to: config.get('meca.notificationEmail'),
-      from: config.get('meca.fromAddressEmail'),
-      subject: 'MECA import failed',
+      to: config.get('meca.email.recipient'),
+      from: config.get('meca.email.sender'),
+      subject: `${config.get('meca.email.subjectPrefix')}MECA import failed`,
```
Comment (diversemix) [inline server/meca/routes.js:43]: Pedantic - but maybe put a space between the prefix and the the "MECA"?

File: server/xpub-server/entities/manuscript/resolvers/submitManuscript.js [hunk 2]
```diff
@@ -55,9 +55,9 @@ async function submitManuscript(_, { data }, { user, ip }) {
         Manuscript.statuses.MECA_EXPORT_FAILED,
       )
       return mailer.send({
-        to: config.get('meca.notificationEmail'),
-        from: config.get('meca.fromAddressEmail'),
-        subject: 'MECA export failed',
+        to: config.get('meca.email.recipient'),
+        from: config.get('meca.email.sender'),
+        subject: `${config.get('meca.email.subjectPrefix')}MECA export failed`,
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/submitManuscript.js:60]: ditto re: space

**Review by diversemix — APPROVED** [review]
Looks good - just pop a space in the subject line. I guess you'll need to remove it from the config too.

**Review by jyenes — APPROVED** [review]
LGTM

File: server/meca/routes.js (hunk 1 above)
Comment (Cooryd) [reply server/meca/routes.js:43]: The space is in the config at the moment:
https://github.com/elifesciences/elife-xpub/blob/e59a7542692f7561126c3b29b37b26f48fc820c3/config/staging.js#L23

I think this is allow no prefix to be defined without having a leading space.

Comment (diversemix) [reply server/meca/routes.js:43]: In fact do we need this ... wouldn't it just make sense to pull it from `NODE_CONFIG_ENV` ?

Comment (diversemix) [reply server/meca/routes.js:43]: Then all the formatting (including the square braces) can be done in the subject line.

**Comment by diversemix** [comment]
@damnhipster - happy for this to go in 'as-is'

File: server/meca/routes.js (hunk 1 above)
Comment (hem-brahmbhatt) [reply server/meca/routes.js:43]: Addressed by #1541 

There's a test for the subject field [here](https://github.com/elifesciences/elife-xpub/blob/1519-staging-email-refactor/server/xpub-server/entities/manuscript/resolvers/submitManuscript.test.js#L213-L216)

File: server/xpub-server/entities/manuscript/resolvers/submitManuscript.js (hunk 2 above)
Comment (hem-brahmbhatt) [reply server/xpub-server/entities/manuscript/resolvers/submitManuscript.js:60]: Addressed by #541

**Pull request merged**